resolver/resolver.go

```diff
@@ -57,11 +57,16 @@ func New(version string, config records.Config) *Resolver {
 		timeout = time.Duration(config.Timeout) * time.Second
 	}
 
+	r.fwds = make(map[string]exchanger.Forwarder)
 	rs := config.Resolvers
-	if !config.ExternalOn {
+	if config.ExternalOn {
+		for zone, resolvers := range config.ZoneResolvers {
+			r.fwds[zone] = exchanger.NewForwarder(resolvers, exchangers(timeout, "udp", "tcp"))
+		}
+	} else {
 		rs = rs[:0]
 	}
-	r.fwd = exchanger.NewForwarder(rs, exchangers(timeout, "udp", "tcp"))
+	r.fwds[""] = exchanger.NewForwarder(rs, exchangers(timeout, "udp", "tcp"))
```

Comment:
dislike magic constant `""` here. can we extract and call it something like `DefaultZone` or `NoZone` or whatever the semantic is?